File: app/src/main/java/org/kindone/willingtodo/persistence/PersistenceProvider.java
package org.kindone.willingtodo.persistence;

/**
 * Created by kindone on 2016. 12. 20..
 */

public interface PersistenceProvider {
    int getVersion();
    TaskContextPersistenceProvider getTaskContextPersistenceProvider();
    TaskPersistenceProvider getTaskPersistenceProvider();
    ConfigPersistenceProvider getConfigPersistenceProvider();
}

File: app/src/main/java/org/kindone/willingtodo/persistence/PersistenceProvider.kt
package org.kindone.willingtodo.persistence

/**
 * Created by kindone on 2016. 12. 20..
 */

interface PersistenceProvider {
    fun getVersion(): Int
    val taskContextPersistenceProvider: TaskContextPersistenceProvider
    val taskPersistenceProvider: TaskPersistenceProvider
    val configPersistenceProvider: ConfigPersistenceProvider
}
